python/pyarrow/parquet/__init__.py

```diff
@@ -3113,9 +3117,14 @@ def file_visitor(written_file):
         dataset.  The first time each partition directory is encountered
         the entire directory will be deleted.  This allows you to overwrite
         old partitions completely.
+        This option is only supported for use_legacy_dataset=False.
     **kwargs : dict,
-        Additional kwargs for write_table function. See docstring for
-        `write_table` or `ParquetWriter` for more information.
+        When use_legacy_dataset=False, used as additional kwargs for 
+        `pyarrow.dataset.write_dataset` function (See docstring for 
+        `write_dataset` or `ParquetFileFormat` for more information).
```

Comment:
Sorry, my mistake:
https://github.com/apache/arrow/pull/13591#discussion_r919724340

Reply:
how about I change it to 
```
When use_legacy_dataset=False, passed as `file_options` to 
        `pyarrow.dataset.write_dataset` function (See docstring for 
        `write_dataset` or `ParquetFileFormat` for more information).
```
because it's passed as file_options to write_dataset first

Reply:
Linter should also be fixed now. Sorry if I still miss any, I'm using the Web editor.

Reply:
The CI is green, thanks! (one fail is not related).

Regarding the change in this discussion: the kwargs are passed first to `ParquetFileFormat.make_write_options` and then to `write_dataset` so we could also mention both?

Reply:
cc @jorisvandenbossche

Reply:
> the kwargs are passed first to `ParquetFileFormat.make_write_options` and then to `write_dataset` so we could also mention both?

Yes, that sounds good (but I would certainly mention `ParquetFileFormat.make_write_options`, as that is where the actual args are passed to)

Reply:
Actually, looking at our docs of this, `make_write_options` isn't that informative, as that docstrings is non-existing (updating this would also be a welcome improvement!). 
In practice, though, it are the keywords that are available in `pq.write_table`/`ParquetWriter`, and those keywords are documented in those functions. So it's probably more useful to also mention those.

Reply:
great! thank you for your suggestion, it's now committed.
I also simplified the function names in the docstring by removing `pyarrow.` prefix, since it was not present elsewhere in docstrings
I hope this looks good now, but please let me know if any further correction are needed.